src/zenml/integrations/daytona/orchestrators/daytona_orchestrator.py

```diff
@@ -0,0 +1,300 @@
+
+
+class DaytonaOrchestrator(WheeledOrchestrator):
+    """Orchestrator responsible for running pipelines using the Daytona SDK."""
+
+    @property
+    def config(self) -> DaytonaOrchestratorConfig:
+        """Returns the `DaytonaOrchestratorConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(DaytonaOrchestratorConfig, self._config)
+
+    @property
+    def settings_class(self) -> Optional[Type["BaseSettings"]]:
+        """Settings class for the Daytona orchestrator.
+
+        Returns:
+            The settings class.
+        """
+        return DaytonaOrchestratorSettings
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Validates the stack.
+
+        Returns:
+            A StackValidator instance.
+        """
+        return StackValidator(
+            required_components=set(),
+            custom_validation_function=None,
+        )
+
+    @property
+    def root_directory(self) -> str:
+        """Path to the root directory for all files concerning this orchestrator.
+
+        Returns:
+            Path to the root directory.
+        """
+        return os.path.join(
+            io_utils.get_global_config_directory(),
+            "daytona",
+            str(self.id),
+        )
+
+    def prepare_or_run_pipeline(
+        self,
+        deployment: "PipelineDeploymentResponse",
+        stack: "Stack",
+        environment: Dict[str, str],
+    ) -> None:
+        """Prepare and run the pipeline using Daytona SDK.
+
+        Args:
+            deployment: The pipeline deployment to prepare or run.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+
+        Raises:
+            ValueError: If the Daytona API key is not set in the settings.
+        """
+        settings = cast(
+            DaytonaOrchestratorSettings, self.get_settings(deployment)
+        )
+        if not settings.api_key:
+            raise ValueError(
+                "Daytona orchestrator requires `api_key` to be set in the settings."
+            )
+
+        # Initialize Daytona client
+        daytona_config = DaytonaConfig(
+            api_key=settings.api_key or "",
+            server_url=settings.server_url or "https://daytona.work/api",
+            target=settings.target or "us",
+        )
+        daytona = Daytona(daytona_config)
+
+        # Create a workspace for the pipeline
+        orchestrator_run_name = get_orchestrator_run_name(
+            pipeline_name=deployment.pipeline_configuration.name
+        )
+
+        logger.info("Creating Daytona workspace...")
+        workspace_params = CreateWorkspaceParams(
+            language="python",
+            id=orchestrator_run_name,
+            name=orchestrator_run_name,
+            image=settings.image,
+            os_user=settings.os_user,
+            env_vars=settings.env_vars,
```

Comment:
Why do we pass some env vars here, and then have some ugly logic that manually `export A=B` stuff later?